examples/preprocessing/plot_discretization_classification.py

```diff
@@ -0,0 +1,163 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+"""
+======================
+Feature discretization
+======================
+
+A demonstration of feature discretization on synthetic classification datasets.
+Feature discretization decomposes each feature into a set of bins, here
+equally distributed in width. The discrete values are then one-hot encoded,
+and given to a linear classifier. On the two non-linearly separable datasets,
```

Comment:
Hold the reader's hand a bit more: the first two rows represent linearly non-separable datasets (moons and concentric circles) while the third is approximately linearly separable.